azext_iot/operations/hub.py

```diff
@@ -1665,16 +1668,30 @@ def _iot_build_sas_token(
         parse_iot_device_module_connection_string,
     )
 
+    uri = None
+    policy = None
+    key = None
+
+    if device_connection_string:
```

Comment:
Lets add tests to exercise the scenario

Reply:
I've noticed it creates a bunch of regression issues in my az iot cli which I need to look at before I go further with it